servicetalk-examples/http/debugging/src/main/java/io/servicetalk/examples/http/debugging/DebuggingExampleClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2018, 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.examples.http.debugging;
+
+import io.servicetalk.concurrent.api.AsyncContext;
+import io.servicetalk.http.api.HttpClient;
+import io.servicetalk.http.api.HttpExecutionStrategies;
+import io.servicetalk.http.api.HttpRequest;
+import io.servicetalk.http.netty.HttpClients;
+
+import java.util.concurrent.CountDownLatch;
+
+import static io.servicetalk.http.api.HttpSerializationProviders.textDeserializer;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.logging.api.LogLevel.TRACE;
+
+/**
+ * The async "Hello World" example with debugging features enabled.
+ */
+public final class DebuggingExampleClient {
+
+    static {
+        // Enables visibility for all wire log messages
+        System.setProperty("servicetalk.logger.wireLogLevel", "TRACE");
+        // Disables the context associated with individual request/responses to simplify execution tracing
+        AsyncContext.disable();
```

Comment:
Because it's easy to forget to remove this statement after the debugging we may demonstrate it can be protected by the system property. Something like:
```java
static {
    if (Boolean.getBoolean("debuggingMode")) {
        AsyncContext.disable();
    }
}
```

Reply:
It would be good to standardize how we use system properties and/or environment variables for configuration and control of debugging features. It is definitely a useful tool. In other orgs I have used 

`(public|private)` static final boolean SOMETHING_DEBUG = Boolean.getBoolean("package.name.Feature.debug");`

and then triggered all use off of the boolean. One advantage is that it lets the JIT remove dead code and also makes it easy to find all the references.